src/boilerplate.c

```diff
@@ -750,26 +749,28 @@ char *infer_mime_type_from_name(const char *file_path) {
 
 char *dump_stdin_into_a_temp_file() {
     char dirpath[] = "/tmp/wl-copy-buffer-XXXXXX";
-    if (mkdtemp(dirpath) != dirpath) {
+    if (mkdtemp(dirpath) == NULL) {
         perror("mkdtemp");
         exit(1);
     }
     char *original_path = path_for_fd(STDIN_FILENO);
 
     char *res_path = malloc(PATH_MAX + 1);
     memcpy(res_path, dirpath, sizeof(dirpath));
-    strcat(res_path, "/");
+    res_path[sizeof(dirpath)] = '/';
+    res_path[sizeof(dirpath) + 1] = '\0';
 
     if (original_path != NULL) {
         char *name = basename(original_path);
-        strcat(res_path, name);
+        // TODO: this might cause a buffer overflow
+        strcpy(res_path + sizeof(dirpath) + 2, name);
     } else {
-        strcat(res_path, "stdin");
+        strcpy(res_path + sizeof(dirpath) + 2, "stdin");
     }
 
     if (fork() == 0) {
         int fd = creat(res_path, S_IRUSR | S_IWUSR);
-        if (fd < 0) {
+        if (fd == -1) {
```

Comment:
Why is it better this way?

Reply:
Because the man pages says that `creat` returns `-1` on error. Not just any negative number.

Reply:
the manual page says it returns -1 on error and "a non-negative integer representing the lowest numbered unused file descriptor" on success.

Both kinds of checks are according to standard.